common/test/acceptance/pages/studio/settings.py

```diff
@@ -17,25 +17,45 @@ class SettingsPage(CoursePage):
     def is_browser_on_page(self):
         return self.q(css='body.view-settings').present
 
+    def get_elements(self, css_selector):
+        self.wait_for_element_presence(
+            css_selector,
+            'Elements matching "{}" selector are present'.format(css_selector)
+        )
+        results = self.q(css=css_selector)
+        return results
+
+    def get_element(self, css_selector):
+        results = self.get_elements(css_selector=css_selector)
+        return results[0] if results else None
```

Comment:
Because you have a wait_for_element_presence method call in the get_elements method, you will never actually get an empty list in results. The wait_for_element_presence method would have raised a BrokenPromise error.